online-shop/src/app/edit/edit.component.html

```diff
@@ -0,0 +1,24 @@
+<div>
+  <form [formGroup]="editProductForm" *ngIf="editProductForm">
```

Comment:
This form is duplicated. I recommend you to extract this form into a shared component that both the add and edit call, and where you pass the form as input parameter.